infra/NotificationRetriever.php

```diff
@@ -50,13 +50,14 @@ public function __construct()
      */
     public function getPSPReferenceByOrderId($orderId)
     {
+        $orderId = intval($orderId);
         $results = $this->db->executeS(
             sprintf(
                 <<<SQL
-select a.pspreference as pspReference
-from %sadyen_notification a
-inner join %sorders o on a.merchant_reference = o.id_cart
-where o.id_order = intval($orderId)
+select op.transaction_id as pspReference
+from %sorder_payment op
+inner join %sorders o on op.order_reference = o.reference
+where o.id_order = $orderId
```

Comment:
could you please run the Presta validator with this change? I think last time it was throwing a security error if the variable was like this. I don't know if it is smart enough to connect it with line 53.